__tests__/components/FriendRequests.test.tsx

```diff
@@ -0,0 +1,241 @@
+        const button = getByLabelText(/accept friend*/i);
+        fireEvent.click(button);
+        await waitFor(()=>{
+            expect(mockedAxios.post).toHaveBeenCalledWith('/api/friends/accept', expect.anything());
+        });
+    });
+
+    test('when accept friend is clicked, axios should be called with the opts {id: senderId}', async ()=>{
+        const requests = [{senderId:'foo', senderEmail: 'foo@bar.com'}];
+        const {getByLabelText} = render(<FriendRequests incomingFriendRequests={requests} />);
+        const button = getByLabelText(/accept friend*/i);
+        fireEvent.click(button);
+        await waitFor(()=>{
+            expect(mockedAxios.post).toHaveBeenCalledWith(expect.anything(), {id: 'foo'});
+        });
+    });
+
+    test('when accept friend is clicked, axios should be called with the opts {id: senderId}, differnt data',
+        async ()=>{
+        const requests = [{senderId:'bar', senderEmail: 'foo@bar.com'}];
+        const {getByLabelText} = render(<FriendRequests incomingFriendRequests={requests} />);
+        const button = getByLabelText(/accept friend*/i);
+        fireEvent.click(button);
+        await waitFor(()=>{
+            expect(mockedAxios.post).toHaveBeenCalledWith(expect.anything(), {id: 'bar'});
+        });
+    });
+
+    test('if accept friend is clicked, setFriendRequests should be called with every sender id except the current one',
+        async ()=>{
+            const requests = [{senderId:'michael', senderEmail: 'michael@correlone.edu'},
+                {senderId: 'sonny', senderEmail: 'santino@correlone.edu'},  {senderId: 'fredo', senderEmail: 'fredo@correlone.edu'},
+                {senderId: 'tom', senderEmail: 'foo@bar.com'}]
+            const {getByRole} = render(<FriendRequests incomingFriendRequests={requests} />);
+            const button = getByRole('button', {
+                name: /accept friend: fredo@correlone.edu/i
+            });
+
+            fireEvent.click(button);
+
+            await waitFor(()=>{
+                expect( screen.queryByText('fredo@correlone.edu')).not.toBeInTheDocument();
+                expect( screen.queryByText('santino@correlone.edu')).toBeInTheDocument();
+                expect( screen.queryByText('michael@correlone.edu')).toBeInTheDocument();
+
+            });
+        });
+
+    test('when accept friend is clicked, should refresh the page', async ()=>{
+        const spy = jest.fn();
+        (useRouter as jest.Mock).mockReturnValue({
+            refresh: spy, // Mock the refresh function
+        });
+        const requests = [{senderId:'foo', senderEmail: 'foo@bar.com'}];
+        const {getByLabelText} = render(<FriendRequests incomingFriendRequests={requests} />);
+        const button = getByLabelText(/accept friend*/i);
+        fireEvent.click(button);
+        await waitFor(()=>{
+            expect(spy).toHaveBeenCalled();
+        });
+    });
+
+    //start copies
```

Comment:
stray comment - if duplicated work is getting to be that much, consider refactoring the test cases to DRY